feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/input/ui/audiorecorder/AudioRecorder.kt

```diff
@@ -0,0 +1,257 @@
+package com.hedvig.android.odyssey.input.ui.audiorecorder
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.hedvig.android.core.designsystem.component.button.LargeContainedTextButton
+import com.hedvig.android.core.designsystem.component.button.LargeTextButton
+import com.hedvig.android.odyssey.R
+import com.hedvig.android.odyssey.repository.AutomationClaimInputDTO2
+import com.hedvig.odyssey.remote.file.File
+import com.hedvig.odyssey.remote.file.FileContent
+import com.hedvig.odyssey.renderers.audiorecorder.PlaybackWaveForm
+import com.hedvig.odyssey.renderers.audiorecorder.RecordingAmplitudeIndicator
+import com.hedvig.odyssey.renderers.utils.ScreenOnFlag
+import com.hedvig.odyssey.utils.contentType
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import kotlinx.coroutines.launch
+import org.koin.androidx.compose.getViewModel
+
+@Composable
+fun AudioRecorderScreen(
+  questions: List<AutomationClaimInputDTO2.AudioRecording.AudioRecordingQuestion>,
+  onAudioFile: suspend (File) -> Unit,
```

Comment:
A suspend lambda passed in here makes me suspect that we could have this logic live elsewhere and have this composable simply call `onNext(audioFile)` and have that function call this instead. But could be wrong and it may not be possible here of course.

Reply:
Yea this is very much work in progress. We need a way of uploading the file somewhere. Lets keep this as a todo for now.